bindings/java/src/lib.rs

```diff
@@ -221,3 +221,8 @@ fn make_capability<'a>(env: &mut JNIEnv<'a>, cap: Capability) -> Result<JObject<
     )?;
     Ok(capability)
 }
+
+fn jstring_to_string(env: &mut JNIEnv, s: &JString) -> Result<String> {
```

Comment:
Please add comment on how it works and why it's safe.

Reply:
This is unsafe of course. The safety assumption is:

```suggestion
/// # Safety
///
/// The caller must guarantee that the Object passed in is an instance
/// of `java.lang.String`, passing in anything else will lead to undefined behavior.
fn jstring_to_string(env: &mut JNIEnv, s: &JString) -> Result<String> {
```

... and we ensure that all JNI calls passing `String` as stated.

Reply:
Emmm... I think it's unsafe here because we don't check the `JString` is actually a `java.lang.String`, I think we should use the `env.get_string()` instead, @tisonkun what do you think?

Reply:
I have a question, why not just use `env.get_string()`?🤔️

Reply:
We already use `into` to convert the JavaStr to str where:

```rust
impl<'str_ref> From<&'str_ref JNIStr> for Cow<'str_ref, str> {
    fn from(other: &'str_ref JNIStr) -> Cow<'str_ref, str> {
        let bytes = other.to_bytes();
        match from_java_cesu8(bytes) {
            Ok(s) => s,
            Err(e) => {
                debug!("error decoding java cesu8: {:#?}", e);
                String::from_utf8_lossy(bytes)
            }
        }
    }
}
```

do the check.

Now that we're ensuring the passed object is String by code, we don't have to use the checked version and thus save cycles:

```
This function has a large relative performance impact compared to Self::get_string_unchecked. For example it may be about five times slower than get_string_unchecked for very short string. This performance penalty comes from the extra validation performed by this function. If and only if you can guarantee that your obj is of java.lang.String, use Self::get_string_unchecked.
```

This is also recommended for significant performance optimization.